src/Azure.Functions.Cli/Helpers/AzureHelper.cs

```diff
@@ -357,9 +390,75 @@ private static async Task ArmHttpAsync(HttpMethod method, Uri uri, string access
             }
         }
 
-        public static async Task PrintFunctionsInfo(Site functionApp, string accessToken, string managementURL, bool showKeys)
+        private static HttpClient functionAppReadyClient = null;
+
+        internal static async Task WaitUntilFunctionAppReadyAsync(Site functionApp, string accessToken, string managementURL,
+            bool showKeys, HttpMessageHandler messageHandler = null)
         {
-            var functions = await GetFunctions(functionApp, accessToken, managementURL);
+            var masterKey = await GetMasterKeyAsync(functionApp.SiteId, accessToken, managementURL);
+
+            if (functionAppReadyClient == null)
+            {
+                HttpMessageHandler handler = messageHandler ?? new HttpClientHandler();
+                if (StaticSettings.IsDebug)
+                {
+                    handler = new LoggingHandler(handler);
+                }
+
+                functionAppReadyClient = new HttpClient(handler);
+                const string jsonContentType = "application/json";
+                functionAppReadyClient.DefaultRequestHeaders.Add("User-Agent", Constants.CliUserAgent);
+                functionAppReadyClient.DefaultRequestHeaders.Add("Accept", jsonContentType);
+            }
+
+            await RetryHelper.Retry(async () =>
+                  {
+                      functionAppReadyClient.DefaultRequestHeaders.Add("x-ms-request-id", Guid.NewGuid().ToString());
+                      var uri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+                      var request = new HttpRequestMessage()
+                      {
+                          RequestUri = uri,
+                          Method = HttpMethod.Get
+                      };
+
+                      var response = await functionAppReadyClient.SendAsync(request);
+                      ColoredConsole.Write(".");
+
+                      if (response.IsSuccessStatusCode)
+                      {
+                          var json = await response.Content.ReadAsAsync<JToken>();
+                          var processUpTime = json["processUptime"]?.ToObject<int>() ?? 0;
+
+                          // Wait 60sec for the readiness of the worker process rebooting by deployment.
+                          if (processUpTime >= Constants.DefaultWorkerProcessUptimeReadiness)  
```

Comment:
waiting 60secs does seem like a long delay. As Core Tools has knowledge of count of functions deployed , make this check dynamic - instead of relying on processUpTime - if host status returns 200 - make call to get function apps - if response matches list of functions in core tools - no need to wait for 60secs

Reply:
The `processUpTime` spec is decided when I talk with @ahmelsayed, so that I'd like to include him to this conversation.

Do you know where `count of functions deployed` ? How can I get to know the host is ready in this case? the increase of the functions deployed means host is ready?

Reply:
As long as this change is scoped to KubeApp, this is ok for now. We can revisit improvement later.

Reply:
Great! I narrow the scope!